### PR TITLE
Add async supervised

### PR DESCRIPTION
**Pull request by ityonemo**
This PR adds support for bringing your own supervisor, in case anyone wants to run their Tasks inside a Task.Supervisor

### CODE REVIEW

File: lib/phoenix_live_view/async.ex [hunk 1]
```diff
@@ -50,14 +50,20 @@ defmodule Phoenix.LiveView.Async do
...
 
-  defp run_async_task(%Socket{} = socket, key, func, kind) do
+  defp run_async_task(%Socket{} = socket, key, func, kind, opts) do
     if Phoenix.LiveView.connected?(socket) do
       lv_pid = self()
       cid = cid(socket)
-      {:ok, pid} = Task.start_link(fn -> do_async(lv_pid, cid, key, func, kind) end)
+      {:ok, pid} = if supervisor = Keyword.get(opts, :supervisor) do
+        {:ok, pid} = Task.Supervisor.start_child(supervisor, fn -> do_async(lv_pid, cid, key, func, kind) end)
+        Process.link(pid)
```
Comment (josevalim) [inline lib/phoenix_live_view/async.ex:62]: There is a race condition here in that, if the LV dies just after spawning the process, then we won't link and the async task will continue. I suggest moving the link inside the start_child anonymous function. :)

File: lib/phoenix_live_view.ex [hunk 2]
```diff
@@ -2024,12 +2024,16 @@ defmodule Phoenix.LiveView do
          |> assign_async([:profile, :rank], fn -> {:ok, %{profile: ..., rank: ...}} end)}
       end
 
+  ## Options
+
+  * `:supervisor` - allows you to specify a `Task.Supervisor` to supervise the task.
+
   See the moduledoc for more information.
```
Comment (josevalim) [inline lib/phoenix_live_view.ex:2031]:
```suggestion
  See the moduledoc for more information.

  ## Options

    * `:supervisor` - allows you to specify a `Task.Supervisor` to supervise the task.

```

File: lib/phoenix_live_view/async.ex [hunk 3]
```diff
@@ -50,14 +50,20 @@ defmodule Phoenix.LiveView.Async do
...
+        {:ok, pid}
```
Comment (josevalim) [inline lib/phoenix_live_view/async.ex:63]: There is a race condition here in that, if the LV process crashes, the spawned process will continue running. I think it is better to link from the spawned process, as it is less likely to crash as we have control over when it starts running external code:

```suggestion
        Task.Supervisor.start_child(supervisor, fn ->
          Process.link(lv_pid)
          do_async(lv_pid, cid, key, func, kind)
        end)
```

**Review by josevalim — COMMENTED** [review]
LGTM! I have dropped some comments.

**Comment by ityonemo** [comment]
thanks!

**Comment by chrismccord** [comment]
❤️❤️❤️🐥🔥

**Pull request merged**